### PR TITLE
Add File Cache config to User Agent

### PR DESCRIPTION
**Pull request by charith87**
The configs currently have only 2 params of File Cache (Enabled, Random Enabled ) to track Its sent in a bitset fashion to reduce the number of characters as the approach is a temporary one till there's a better way to send configuration details.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - NA

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -70,34 +70,46 @@ func registerSIGINTHandler(mountPoint string) {
...
-		return fmt.Sprintf("gcsfuse/%s (GPN:gcsfuse)", getVersion())
+		return fmt.Sprintf("gcsfuse/%s (GPN:gcsfuse) (Cfg:%s)", getVersion(), configString)
 	}
 }
 
-func createStorageHandle(flags *flagStorage) (storageHandle storage.StorageHandle, err error) {
+func getConfigForUserAgent(mountConfig *config.MountConfig) string {
+	// Minimum configuration details created in a bitset fashion. Right now, its restricted only to File Cache Settings.
+	isFileCacheEnabled := "0"
+	if mountConfig.FileCacheConfig.MaxSizeMB != 0 && string(mountConfig.CacheDir) != "" {
```
Comment (sethiay) [inline main.go:88]: create common function for checking if cache enabled in the config_util https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/config/mount_config.go

File: main_test.go [hunk 2]
```diff
@@ -46,32 +47,81 @@ func (t *MainTest) TestGetUserAgentWhenMetadataImageTypeEnvVarIsSet() {
...
+	mountConfig3 := &config.MountConfig{
+		FileCacheConfig: config.FileCacheConfig{
+			MaxSizeMB: -1,
+		},
+	}
+	userAgent3 := getUserAgent("AppName", getConfigForUserAgent(mountConfig3))
+	expectedUserAgent3 := strings.TrimSpace(fmt.Sprintf("gcsfuse/%s (GPN:gcsfuse-AppName) (Cfg:00)", getVersion()))
+	ExpectEq(expectedUserAgent3, userAgent3)
+
+	// Test File Cache disabled when Cache Dir is given but maxSize is not set.
```
Comment (sethiay) [inline main_test.go:106]: use 0 instead of "set"

File: main.go [hunk 3]
```diff
@@ -70,34 +70,46 @@ func registerSIGINTHandler(mountPoint string) {
...
+		isFileCacheEnabled = "1"
+	}
+	isFileCacheForRangeReadEnabled := "0"
+	if mountConfig.FileCacheConfig.CacheFileForRangeRead {
+		isFileCacheForRangeReadEnabled = "1"
+	}
+	return fmt.Sprintf("%s%s", isFileCacheEnabled, isFileCacheForRangeReadEnabled)
```
Comment (sethiay) [inline main.go:95]: add delimiter (:) between the config elements

File: main.go [hunk 4]
```diff
@@ -70,34 +70,46 @@ func registerSIGINTHandler(mountPoint string) {
 	}()
 }
 
-func getUserAgent(appName string) string {
+func getUserAgent(appName string, configString string) string {
```
Comment (sethiay) [inline main.go:73]: nit: remove "String"

File: main.go (hunk 1 above)
Comment (charith87) [reply main.go:88]: Done.

File: main_test.go (hunk 2 above)
Comment (charith87) [reply main_test.go:106]: Done.

File: main.go (hunk 4 above)
Comment (charith87) [reply main.go:73]: done.

File: main.go (hunk 3 above)
Comment (charith87) [reply main.go]: done

File: main_test.go [hunk 5]
```diff
@@ -46,32 +47,81 @@ func (t *MainTest) TestGetUserAgentWhenMetadataImageTypeEnvVarIsSet() {
...
+func (t *MainTest) TestGetUserAgentConfig() {
 
+	// Test Default Case - No File Cache
+	mountConfig := &config.MountConfig{}
+	userAgent := getUserAgent("AppName", getConfigForUserAgent(mountConfig))
+	expectedUserAgent := strings.TrimSpace(fmt.Sprintf("gcsfuse/%s (GPN:gcsfuse-AppName) (Cfg:0:0)", getVersion()))
 	ExpectEq(expectedUserAgent, userAgent)
+
+	// Test File Cache Enabled and Random Read Enabled
+	mountConfig1 := &config.MountConfig{
```
Comment (sethiay) [inline main_test.go:74]: would prefer separating the unit tests like TestGetUserAgentWhenFileCacheIsEnabled

File: internal/config/config_util_test.go [hunk 6]
```diff
@@ -88,3 +88,32 @@ func (t *ConfigTest) TestOverrideLoggingFlags_WithEmptyLogConfigs() {
 	AssertEq(mountConfig.LogConfig.FilePath, "a.txt")
 	AssertEq(mountConfig.LogConfig.Severity, INFO)
 }
+
+func (t *ConfigTest) TestIsReadCacheEnabled() {
```
[the remark this answers is not in the document]
Comment (charith87) [reply internal/config/config_util_test.go]: Done.

File: main_test.go (hunk 5 above)
Comment (charith87) [reply main_test.go]: Done. Kindly review.

**Review by sethiay — APPROVED** [review]

**Pull request merged**